src/charm.py

```diff
@@ -117,28 +119,36 @@ def _on_config_changed(self, event: ops.ConfigChangedEvent) -> None:
                 return
             self._stored.binary_url = self.config.get('binary-url')
             self._stored.docker_tag = self.config.get('docker-tag')
+            update_service_args = True
 
         # Update of polkadot service arguments requested
         if self._stored.service_args != self.config.get('service-args'):
             self.unit.status = ops.MaintenanceStatus("Updating service args")
-            utils.update_service_args(service_args_obj.service_args_string)
             self._stored.service_args = self.config.get('service-args')
+            update_service_args = True
 
         if self._stored.chain_spec_url != self.config.get('chain-spec-url'):
             self.unit.status = ops.MaintenanceStatus("Updating chain spec")
-            utils.update_service_args(service_args_obj.service_args_string)
             self._stored.chain_spec_url = self.config.get('chain-spec-url')
+            update_service_args = True
 
         if self._stored.local_relaychain_spec_url != self.config.get('local-relaychain-spec-url'):
             self.unit.status = ops.MaintenanceStatus("Updating relaychain spec")
-            utils.update_service_args(service_args_obj.service_args_string)
             self._stored.local_relaychain_spec_url = self.config.get('local-relaychain-spec-url')
+            update_service_args = True
 
         if self._stored.wasm_runtime_url != self.config.get('wasm-runtime-url'):
             self.unit.status = ops.MaintenanceStatus("Updating wasm runtime")
             utils.download_wasm_runtime(self.config.get('wasm-runtime-url'))
-            utils.update_service_args(service_args_obj.service_args_string)
             self._stored.wasm_runtime_url = self.config.get('wasm-runtime-url')
+            update_service_args = True
+
+        if update_service_args:
+            if not utils.service_started(): 
+                utils.update_service_args(service_args_obj.service_args_string)            
+            else:
+                self.unit.status = ops.MaintenanceStatus("Service running")
+                return
```

Comment:
So what this does now is:
- If `update_service_args` - go deeper
  - If service is stopped, update the service args 
  - If service is running, set status `"Service running"`

Seems to me this isn't correct either. Shouldn't it be:

```suggestion
        if update_service_args:
            utils.update_service_args(service_args_obj.service_args_string)   
            if utils.service_started(): 
                utils.restart_service()
```

- No `return`, because we want the function to move onto `update_status_simple` no matter what happens here.
- Don't set any status, that will be done in `update_status_simple`.
- This will need the `update_service_args()` function to change so that it doesn't include a service restart, which will in turn affect other places it is called. I suggest adding an argument `restart_service: bool` to the function and then make the choice individually for each calling function whether it should restart.